pkg/tcpip/network/ipv4/icmp.go

```diff
@@ -134,6 +134,44 @@ func (e *endpoint) handleICMP(r *stack.Route, netHeader buffer.View, vv buffer.V
 	case header.ICMPv4Timestamp:
 		received.Timestamp.Increment()
 
+		// Only send a reply if the checksum is valid.
+		wantChecksum := h.Checksum()
+
+		// It's possible that a raw socket expects to receive this.
```

Comment:
why? the line above just read the Checksum field of this header into wantChecksum and this just sets it right back?

Reply:
Actually below that, need to check the if the got checksum is equal to want checksum or not. that's why we are storing the checksum value to **wantChecksum** before we are setting it to zero.